include/thread_pool/thread_pool.hpp

```diff
@@ -150,17 +186,17 @@ namespace tp
     {
         const auto ok = try_post(std::forward<Handler>(handler));
         assert(ok);
+        _tp_unused(ok);
     }
 
     template <typename TSettings>
-    inline Worker& ThreadPool<TSettings>::getWorker()
+    inline typename ThreadPool<TSettings>::FixedWorker& ThreadPool<TSettings>::getWorker()
```

Comment:
I can understand that.  If you both feel these changes aren't compatible with the project objectives, I can maintain a `c++1y` fork as needed.